arroyo/processing/strategies/streaming/transform.py

```diff
@@ -293,34 +318,35 @@ def __init__(
         self.__shared_memory_manager.start()
 
         self.__pool = Pool(
-            processes,
+            num_processes,
             initializer=partial(parallel_transform_worker_initializer, initializer),
             context=multiprocessing.get_context("spawn"),
         )
 
         self.__input_blocks = [
             self.__shared_memory_manager.SharedMemory(input_block_size)
-            for _ in range(processes)
+            for _ in range(num_processes)
         ]
 
         self.__output_blocks = [
             self.__shared_memory_manager.SharedMemory(output_block_size)
-            for _ in range(processes)
+            for _ in range(num_processes)
         ]
 
         self.__batch_builder: Optional[BatchBuilder[TPayload]] = None
 
-        self.__results: Deque[
+        self.__processes: Deque[
             Tuple[
                 MessageBatch[TPayload],
-                AsyncResult[Tuple[int, MessageBatch[TTransformed]]],
+                AsyncResult[ParallelTransformResult[TTransformed]],
+                MutableSequence[InvalidMessage],
             ]
```

Comment:
I tired to pull this main `Tuple` out into a dataclass like I did for the result, but for some reason I can't add generic type params to `AsyncResult` when it's part of a class.

Eg.
```python3
@dataclass
class ParallelProcess:
    async_result: AsyncResult[<some_type>]
```
raises this:
```python3
TypeError: 'type' object is not subscriptable
```

Reply:
I have no idea, if I don't add generic types mypy complains about it. These types also help later on cause we know what type would result from the async job. Maybe there is another "type" that is meant to be used for an AsyncResult/ApplyResult?

Reply:
Yeah, no need to change this. Just confusing how this is working.